### PR TITLE
Ports TGChat

### PR DESCRIPTION
**Pull request by SplinterGP**
## About The Pull Request
Ports [TGChat](https://github.com/tgstation/tgstation/pull/52426) that uses TGUI.

This PR should be quite smaller than TGUI 4.3 port PR since TGUI PR already brought tgchat stuff, this is basically removing goonchat from code, adding few tgchat and changing some stuff on already on /tgui folder like the SCSS

Still has some stuff to finish, like doing the light mode SCSS and some admin to_chat() changes.

this is how i've got it so far, fixed a few things you might see wrong on it on photo:

image

## Why It's Good For The Game
Lighter than goonchat, reduces server load and its better overall, and carries past chat messages between connects(And through all servers that uses TGChat, since its a cache file on your PC).

## Changelog
:cl:
add: Added TGChat in place of Goonchat.
del: Goonchat removed.
code: updated icon2base64 proc to not use a global file that doens't exist anymore.
/:cl:

### CODE REVIEW

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -878,7 +920,10 @@
 	var/keywordparsedmsg = keywords_lookup(msg)
 
 	if(external)
-		to_chat(src, "<span class='notice'>PM to-<b>Staff</b>: <span class='linkify'>[rawmsg]</span></font>")
+		to_chat(src,
+			type = MESSAGE_TYPE_ADMINPM,
+			html = "<span class='notice'>PM to-<b>Staff</b>: <span class='linkify'>[rawmsg]</span></font>",
+			confidential = TRUE)
```
Comment (TiviPlus) [inline code/modules/admin/admin_verbs.dm:926]: Delete all the confidential junk, it was for Replays IIRC and we dont use it

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Did you port the reconnect issue workaround?

**Comment by SplinterGP** [comment]
> Did you port the reconnect issue workaround?

ye

File: code/modules/admin/admin_verbs.dm [hunk 2]
```diff
@@ -480,8 +480,7 @@
 		if(check_other_rights(C, R_ASAY, FALSE))
 			to_chat(C,
 				type = MESSAGE_TYPE_ADMINCHAT,
-				html = msg,
-				confidential = TRUE)
+				html = msg,)
```
Comment (TiviPlus) [inline code/modules/admin/admin_verbs.dm:483]: trailing comma

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/admin/admin_verbs.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/admin/admin_verbs.dm:483]:
```suggestion
				html = msg)
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**